File: csharp/TDD/SimpleBookPricer.cs
using System;

public static class SimpleBookPricer
{
    public static double CalculatePrice(BookType type, double rrp, int monthsSincePublication)
    {
        if (type == BookType.Paperback)
        {
            return rrp;
        }

        if (monthsSincePublication > 6)
        {
            return Math.Round(rrp / 2.0);
        }

        double discount = 0;

        if (monthsSincePublication == 0)
        {
            discount = 2;
        }

        return rrp - discount;
    }
}

File: python/tdd/simple_book_pricer.py
def calculate_price(title, author, book_type, rrp, month):
    if book_type == "PAPERBACK":
        return rrp
    elif book_type == "HARDBACK":
        if month == 0:
            return rrp - 2
        elif month >= 1 and month <= 6:
            return rrp
        else:
            return rrp * 0.5
    else:
        raise ValueError("Invalid book type")
